dhall/tests/Dhall/Test/TypeInference.hs

```diff
@@ -84,25 +84,32 @@ successTest prefix = do
         Tasty.HUnit.assertEqual message resolvedExpectedType inferredType
 
 failureTest :: Text -> TestTree
-failureTest path = do
-    let skip = [ typeInferenceDirectory </> "failure/unit/MergeEmptyNeedsDirectAnnotation1.dhall"
+failureTest prefix = do
+    let skip = [ typeInferenceDirectory </> "failure/unit/MergeEmptyNeedsDirectAnnotation1"
                ]
 
-    Test.Util.testCase path skip $ do
-        let dhallPath = Test.Util.toDhallPath path
+    Test.Util.testCase prefix skip $ do
+        let prefixFP = Text.unpack prefix
 
-        expression <- Core.throws (Parser.exprFromText mempty dhallPath)
+        code <- Text.IO.readFile (prefixFP <> ".dhall")
 
-        let io :: IO Bool
-            io = do
-                _ <- Test.Util.load expression
-                return True
+        case Parser.exprFromText mempty code of
+            Left _ -> return ()
+                      -- Duplicate fields are incorrectly caught during parsing:
+                      -- https://github.com/dhall-lang/dhall-haskell/issues/772
 
-        let handler :: SomeException -> IO Bool
-            handler _ = return False
+            Right expression -> do
+                tryResolvedExpr <-
+                    Exception.try
+                       (Test.Util.loadRelativeTo
+                           (FilePath.takeDirectory prefixFP)
+                           Import.IgnoreSemanticCache
+                           expression)
 
-        typeChecked <- Exception.handle handler io
+                case tryResolvedExpr of
+                    Left (SomeException _) -> return ()
```

Comment:
I think it would be better if we wouldn't have to handle errors from import resolution here. This is only for two tests,
`importBoundary` and `customHeadersUsingBoundVariable`. I wonder whether we could put these into the import test suite instead – see https://github.com/dhall-lang/dhall-lang/issues/748.